service/resource/persistentvolume/update.go

```diff
@@ -34,13 +35,66 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateState inter
 		return nil
 	}
 
-	_, err = toPV(obj)
+	pv, err := toPV(obj)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	// update state logic
+	switch combinedState := string(rpv.State) + rpv.RecycleState; combinedState {
+	case "ReleasedRecycled":
+		pv, err := r.newRecycleStateAnnotation(pv, released)
```

Comment:
Won't it be just deleted in next resyncperiod?

Reply:
Yes, but that is a way to follow volume state changes in this flow. Because if I skip this stage - I'll need to return default recycle state as `Released` for never cleaned volumes before. And that breaks a logic a bit

Reply:
For me it's rather a question why not delete immediately?

Reply:
I didn't delete it immediately because there was a case when a volume was never cleaned before and it has no any recycle state. But it turned out also that I can do it in one state, so `Released` recycle state become useless. Updated